File: backend/app/utils/demo_accounts.py
"""Shared demo accounts — OTP verification is skipped for these emails or phones."""

DEMO_EMAILS: frozenset[str] = frozenset(
    {
        "admin@washgo.demo",
        "customer@washgo.demo",
        "partner@washgo.demo",
    }
)

DEMO_PHONES: frozenset[str] = frozenset(
    {
        "9876543210",  # admin
        "9876543211",  # customer
        "9876543212",  # partner
    }
)


def _normalize_phone_digits(phone: str) -> str:
    import re

    digits = re.sub(r"\D", "", phone or "")
    if digits.startswith("91") and len(digits) == 12:
        digits = digits[2:]
    return digits


def is_demo_email(email: str) -> bool:
    return email.strip().lower() in DEMO_EMAILS


def is_demo_phone(phone: str) -> bool:
    return _normalize_phone_digits(phone) in DEMO_PHONES


def is_demo_account(*, email: str | None = None, phone: str | None = None) -> bool:
    if email and is_demo_email(email):
        return True
    if phone and is_demo_phone(phone):
        return True
    return False

File: frontend/src/lib/demoAccounts.js
import { normalizeIndianPhoneDigits } from '../utils/validators';

/** Must match backend `app.utils.demo_accounts`. */
export const ADMIN_DEMO_EMAIL = 'admin@washgo.demo';
export const ADMIN_DEMO_PHONE = '9876543210';

export const DEMO_EMAILS = [ADMIN_DEMO_EMAIL, 'customer@washgo.demo', 'partner@washgo.demo'];

export const DEMO_PHONES = ['9876543210', '9876543211', '9876543212'];

export const DEMO_ACCOUNTS = [
  { label: 'Admin', phone: '9876543210', email: 'admin@washgo.demo', where: '/admin/login' },
  { label: 'Customer', phone: '9876543211', email: 'customer@washgo.demo', where: '/login' },
  { label: 'Partner', phone: '9876543212', email: 'partner@washgo.demo', where: '/partner/login' },
];

export const DEMO_PASSWORD = 'Demo1234';

export function isDemoEmail(email) {
  return DEMO_EMAILS.includes(String(email || '').trim().toLowerCase());
}

export function isAdminDemoEmail(email) {
  return String(email || '').trim().toLowerCase() === ADMIN_DEMO_EMAIL;
}

export function isDemoPhone(phone) {
  return DEMO_PHONES.includes(normalizeIndianPhoneDigits(phone));
}

export function isAdminDemoPhone(phone) {
  return normalizeIndianPhoneDigits(phone) === ADMIN_DEMO_PHONE;
}

export function isDemoAccount({ email, phone } = {}) {
  if (email && isDemoEmail(email)) return true;
  if (phone && isDemoPhone(phone)) return true;
  return false;
}

export const ADMIN_LOGIN_ONLY_MESSAGE =
  'Administrator accounts must sign in at the admin console (/admin/login), not here.';
